# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: dom/serviceworkers/FetchEventOpParent.cpp
# Commit: 8fec60d0c499
# Full Hash: 8fec60d0c499fdf429f4d11da6bea0c65aecb1d4
# Author: Eden Chuang <echuang@mozilla.com>
# Date: 2022-02-27 09:40:54
# Regressor Bug: 1754365
# File Overlap Count: 1
# Description:
#   Bug 1754365 - Create IPC for transferring NavigationPreload's PerformanceTimingData. r=dom-worker-reviewers,jesup
#   
#   Depends on D138250
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D138251
# ==============================================================================

diff -r fb0ac558803d -r 8fec60d0c499 dom/serviceworkers/FetchEventOpParent.cpp
--- a/dom/serviceworkers/FetchEventOpParent.cpp	Sun Feb 27 00:26:21 2022 +0000
+++ b/dom/serviceworkers/FetchEventOpParent.cpp	Sun Feb 27 00:26:22 2022 +0000
@@ -25,9 +25,9 @@
 
 namespace dom {
 
-Maybe<ParentToParentInternalResponse> FetchEventOpParent::OnStart(
+Maybe<ParentToParentResponseWithTiming> FetchEventOpParent::OnStart(
     MovingNotNull<RefPtr<FetchEventOpProxyParent>> aFetchEventOpProxyParent) {
-  Maybe<ParentToParentInternalResponse> preloadResponse =
+  Maybe<ParentToParentResponseWithTiming> preloadResponse =
       std::move(mState.as<Pending>().mPreloadResponse);
   mState = AsVariant(Started{std::move(aFetchEventOpProxyParent)});
   return preloadResponse;
@@ -39,7 +39,7 @@
 }
 
 mozilla::ipc::IPCResult FetchEventOpParent::RecvPreloadResponse(
-    ParentToParentInternalResponse&& aResponse) {
+    ParentToParentResponseWithTiming&& aResponse) {
   AssertIsOnBackgroundThread();
 
   mState.match(
@@ -48,11 +48,17 @@
         aPending.mPreloadResponse = Some(std::move(aResponse));
       },
       [&aResponse](Started& aStarted) {
+        ParentToChildResponseWithTiming response;
         auto backgroundParent = WrapNotNull(
             WrapNotNull(aStarted.mFetchEventOpProxyParent->Manager())
                 ->Manager());
+        response.response() =
+            ToParentToChild(aResponse.response(), backgroundParent);
+        response.timingData() = aResponse.timingData();
+        response.initiatorType() = aResponse.initiatorType();
+        response.entryName() = aResponse.entryName();
         Unused << aStarted.mFetchEventOpProxyParent->SendPreloadResponse(
-            ToParentToChild(aResponse, backgroundParent));
+            response);
       },
       [](const Finished&) {});
 